app/controllers/candidate_interface/offer_dashboard_controller.rb

```diff
@@ -0,0 +1,13 @@
+module CandidateInterface
+  class OfferDashboardController < CandidateInterfaceController
+    def show
+      @application_form = current_application
+      @application_choice_with_offer = current_application.application_choices.pending_conditions.first
+      @accepted_offer_provider_name = @application_choice_with_offer.provider.name
+    end
+
+    def view_reference
+      @reference = current_application.application_references.find(params[:id])
```

Comment:
Do we handle `params[:id]` being invalid or not being found here? Is this handled by a render 404 response or does it raise an exception?

Reply:
It will raise an `ActiveRecord::RecordNotFound`

Reply:
Do you think we need to?

```ruby
rescue ActiveRecord::RecordNotFound
  render_404
```

Reply:
Yeah I'd say if we don't do this in a base controller, [and judging by these sort of errors we don't](https://sentry.io/organizations/dfe-teacher-services/issues/3070145191/?referrer=slack) we should respond with 404 for any non-existent or nonsensical `id` coming from params/user input. Otherwise the user will see a 500 and we get Sentry errors like the references one.